File: dofbot_c++/beep.cpp
/**
 * @file beep.cpp
 * @brief Demonstrates the buzzer control commands.
 */

#include "Arm_Lib.h"
#include "cli_args.h"

#include <chrono>
#include <exception>
#include <iostream>
#include <thread>

int main(int argc, char* argv[]) {
    const std::string description =
        "Buzzer demo for the Dofbot arm. Plays a few tones to confirm communication.";

    try {
        const CommonArgs args = parse_common_args(argc, argv, description);
        Arm_Device arm(args.port);

        std::this_thread::sleep_for(std::chrono::duration<double>(args.init_delay));

        std::cout << "Short beep..." << std::endl;
        arm.Arm_Buzzer_On(1);
        std::this_thread::sleep_for(std::chrono::seconds(1));

        std::cout << "Longer beep..." << std::endl;
        arm.Arm_Buzzer_On(3);
        std::this_thread::sleep_for(std::chrono::seconds(1));

        std::cout << "Continuous tone..." << std::endl;
        arm.Arm_Buzzer_On();
        std::this_thread::sleep_for(std::chrono::seconds(1));

        std::cout << "Silencing buzzer." << std::endl;
        arm.Arm_Buzzer_Off();
        std::this_thread::sleep_for(std::chrono::seconds(1));

    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << std::endl;
        return 1;
    }

    return 0;
}

File: dofbot_python/beep.py
"""Buzzer demo for the Dofbot arm.

Runs a short sequence that exercises the onboard buzzer with different
durations so team members can quickly confirm the arm is responsive.
"""

from Arm_Lib import Arm_Device
import time

from cli_args import parse_common_args


def main() -> None:
    args = parse_common_args(__doc__)
    arm = Arm_Device(args.port)
    try:
        time.sleep(args.init_delay)

        # Brief beep to verify hardware responds to commands.
        arm.Arm_Buzzer_On(1)
        time.sleep(1)

        # Slightly longer chirp for comparison.
        arm.Arm_Buzzer_On(3)
        time.sleep(1)

        # Hold the buzzer on until we explicitly switch it off.
        arm.Arm_Buzzer_On()
        time.sleep(1)

        # Silence the buzzer before exiting so the arm is quiet.
        arm.Arm_Buzzer_Off()
        time.sleep(1)
    finally:
        del arm


if __name__ == "__main__":
    main()
